### PR TITLE
Adding Inject UnloggedRunnerTest file pop-up functionality

### PR DESCRIPTION
**Pull request by kingkong-AI**
**Change Description:**

This change creates a Pop-up to inject UnloggedTest.java File into the project.

- Has option to copy the code from the text box.
- Inject the code directly into the respective module selected from the drop down.
- Open link to the documentation.

image

**Testing**

Manually tested the working of the features on different projects on both Mac and Windows:

1. _maven single-module project_
2. _maven multi-module project_
3. _gradle single-module project_
4. _gradle multi-module project_
5. _maven-gradle mixed module project_

### CODE REVIEW

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
+        } else if(type.equals(CopyType.MAVEN)){
+            insidiousService.copyToClipboard(mvnTestTextArea.getText());
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
+        } else {
+            insidiousService.copyToClipboard(gradleTestTextArea.getText());
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
```
Comment (TheAmg) [inline src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java:131]: Call to `notifyMessage()` can be moved outside of conditional blocks.

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+     *
+     * @return the content string for the test runner file
+     */
+    private String generateRunnerFileContent() {
+        return "import io.unlogged.runner.UnloggedTestRunner;\n" +
+                "import org.junit.runner.RunWith;\n\n" +
+                "@RunWith(UnloggedTestRunner.class)\n" +
+                "public class UnloggedTest {\n" +
+                "}\n";
+    }
```
Comment (TheAmg) [inline src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java:132]: This code is present in 2 places, here and `TestRunnerInjector`. They could be managed from a single source.

**Review by TheAmg — CHANGES_REQUESTED** [review]

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java (hunk 1 above)
Comment (kingkong-AI) [reply src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java:131]: will take it up with other changes in the revision

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java (hunk 2 above)
Comment (kingkong-AI) [reply src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java:132]: will take this up with other changes in the revision

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+     */
+    public JPanel getMainPanel() {
+        return mainPanel;
+    }
+
+    /**
+     * Copies the code to the clipboard based on the specified CopyType.
+     * @param type the type of code to copy (RUNNER, MAVEN, or GRADLE)
+     */
+    public void copyCode(CopyType type) {
```
Comment (kingkong-AI) [inline src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java:119]: making these methods private

File: src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java (hunk 2 above)
Comment (kingkong-AI) [reply src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java:132]: created CommonUtil class that will have a function generateRunnerFileContent()

**Pull request merged**